iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -118,18 +120,23 @@ public void commitTask(TaskAttemptContext originalContext) throws IOException {
           .run(output -> {
             Table table = HiveIcebergStorageHandler.table(context.getJobConf(), output);
             if (table != null) {
-              HiveIcebergRecordWriter writer = writers.get(output);
-              DataFile[] closedFiles;
+              HiveIcebergWriter writer = writers.get(output);
+              HiveIcebergWriter delWriter = delWriters.get(output);
+              String fileForCommitLocation = generateFileForCommitLocation(table.location(), jobConf,
+                  attemptID.getJobID(), attemptID.getTaskID().getId());
+              if (delWriter != null) {
+                DeleteFile[] closedFiles = delWriter.deleteFiles().toArray(new DeleteFile[0]);
+                createFileForCommit(closedFiles, fileForCommitLocation, table.io());
```

Comment:
would it make sense to write into a single `forCommit` file?

Reply:
We could write into a single forCommit, but it would complicate things during the jobCommit operation. We'd have to read the files back from disk and check for each file separately whether it's a DataFile or a DeleteFile, since they each need a different Iceberg API call to commit them (`table.newAppend()` vs `table.newRowDelta()`, respcetively)

Reply:
Maybe we can create a little bit more complex data structure to serialise.
I think creating/reading back the S3 files is where we will spend some serious time

Reply:
>  the S3 files is where we will spend some serious time

Makes sense.
As discussed, let's create a container object which we can serialize/deserialize

Reply:
I've created a new container class `FilesForCommit`, which we now use to serialize into S3 during commitTask, and read it back during jobCommit